samples/web/content/apprtc/room.py

```diff
@@ -65,6 +65,27 @@ def has_room(host, room_id):
   room = memcache_client.get(key)
   return room is not None
 
+def remove_room(host, room_id, client_id):
+  memcache_client = memcache.Client()
```

Comment:
A should not create a new room if A is a callee.

On Fri, Jan 30, 2015 at 1:12 PM, Chuck Hays <notifications@github.com>
wrote:

> In samples/web/content/apprtc/room.py
> <https://github.com/GoogleChrome/webrtc/pull/456#discussion_r23872714>:
>
> > @@ -65,6 +65,27 @@ def has_room(host, room_id):
> >    room = memcache_client.get(key)
> >    return room is not None
> >
> > +def remove_room(host, room_id, client_id):
> > +  memcache_client = memcache.Client()
>
> I hadn't thought about it in terms of the actual caller/callee, just the
> race condition in the code.
>
> I think A and B would both be callees. A is from accept, B is from decline.
>
> I'll think this through a little more. It might make sense to move the
> Room creation to a more explicit call instead of being a side effect of the
> add_client update process.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/456/files#r23872714>.
>



-- 
Jiayang Liu